cmd/collector/main.go

```diff
@@ -195,13 +198,33 @@ func startGRPCServer(
 		if opts.CollectorGRPCCert == "" || opts.CollectorGRPCKey == "" {
 			return nil, fmt.Errorf("you requested TLS but configuration does not include a path to cert and/or key")
 		}
-		creds, err := credentials.NewServerTLSFromFile(
-			opts.CollectorGRPCCert,
-			opts.CollectorGRPCKey,
-		)
+
+		tlsCfg := &tls.Config{
```

Comment:
this code should not be in main, but somewhere in the server factory, maybe similar to `grpcserver.StartGRPCCollector()` function. Right now it appears that similar code is repeated in the unit tests. Can you move it to `grpcserver` and reuse in unit tests so that it's also tested? We can't really test main()